File: edge-plugin/src/main/scala/com/cloudentity/edge/plugin/ValidatePluginService.java
package com.cloudentity.edge.plugin;

import com.cloudentity.edge.plugin.config.*;
import com.cloudentity.tools.vertx.bus.VertxEndpoint;
import io.vertx.core.Future;

public interface ValidatePluginService {
  @VertxEndpoint(address = ":plugin.validate")
  Future<ValidateResponse> validateConfig(ValidateRequest req);
}

File: edge-plugin/src/main/scala/com/cloudentity/edge/plugin/ValidatePluginService.scala
package com.cloudentity.edge.plugin

import com.cloudentity.edge.plugin.config._
import com.cloudentity.tools.vertx.bus.VertxEndpoint
import io.vertx.core.Future

trait ValidatePluginService {
  @VertxEndpoint(address = ":plugin.validate")
  def validateConfig(req: ValidateRequest): Future[ValidateResponse]
}
